app/recipeapp.js

```diff
@@ -10,94 +10,35 @@ class Recipe {
-      "Thyme",
-      "Bay leaf",
-      "Beef broth",
-      "Butter",
-      "Olive oil",
-      "Salt",
-      "Pepper",
-    ],
-    "https://www.telegraph.co.uk/content/dam/recipes/2021/01/05/beef-soup_trans_NvBQzQNjv4Bq_nNpUE6IaHI6R9FXzzuwlx_fMl3rEm0l9EtNXQnfBIo.png" // Image URL for Boeuf Bourguignon
-  ),
-  new Recipe(
-    1002,
-    "Ratatouille",
-    "A vegetable stew from Provence, France. Traditionally made with eggplant, zucchini, tomatoes, peppers, onions, and garlic.",
-    [
-      "Eggplant",
-      "Zucchini",
-      "Tomatoes",
-      "Peppers",
-      "Onions",
-      "Garlic",
-      "Olive oil",
-      "Herbs (such as thyme, basil, and bay leaf)",
-      "Salt",
-      "Pepper",
-    ],
-    "https://www.sprinklesandsprouts.com/wp-content/uploads/2022/04/RatatouilleSQ.jpg" // Image URL for Ratatouille
-  ),
-  new Recipe(
-    1003,
-    "Soupe à l'Oignon Gratinée (French Onion Soup)",
-    "A rich and flavorful soup made with caramelized onions, beef broth, and croutons, topped with melted cheese.",
-    [
-      "Onions (usually yellow onions)",
-      "Beef broth",
-      "Red wine (optional)",
-      "Butter",
-      "Olive oil",
-      "Flour",
-      "Thyme",
-      "Bay leaf",
-      "Salt",
-      "Pepper",
-      "Baguette",
-      "Gruyère cheese",
-    ],
-    "https://www.fifteenspatulas.com/wp-content/uploads/2011/10/French-Onion-Soup-Fifteen-Spatulas-12.jpg" // Image URL for French Onion Soup
-  ),
-
-  new Recipe(
-    1004,
-    "Quiche Lorraine",
-    "A savory tart with a flaky crust, filled with eggs, cheese, and bacon (can be made vegetarian by omitting bacon).",
-    [
-      "Flour",
-      "Butter",
-      "Eggs",
-      "Heavy cream",
-      "Cheese (Gruyère or Emmental)",
-      "Bacon (optional)",
-      "Onions",
-      "Nutmeg",
-      "Salt",
-      "Pepper",
-    ],
-    "https://img.taste.com.au/yzM49jFu/taste/2017/02/classic-quiche-lorraine-121391-2.jpg" // Image URL for Quiche Lorraine
-  ),
-
-  new Recipe(
-    1005,
-    "Crêpes",
-    "French pancakes made from flour, eggs, and milk. Can be enjoyed savory or sweet with various fillings.",
-    ["Flour", "Eggs", "Milk", "Water", "Butter", "Salt", "Sugar (optional)"],
-    "https://www.sweetashoney.co/wp-content/uploads/French-Crepe-Recipe-1.jpg" // Image URL for Crêpes
-  ),
-];
+frenchRecipes = [];
+fetchedRecipes = [];
+ingredientsPrices = {};
+
+async function getData() {
+  const response = await fetch(
+    "https://raw.githubusercontent.com/ItsNaklov/itsnaklov.github.io/main/recipe.json"
+  );
+  const data = await response.json();
+  console.log(data);
+  fetchedRecipes = data["recipes"];
+  ingredientsPrices = data["ingredientsPrices"];
+  fetchedRecipes.forEach((recipie) => {
+    // id, name, description, ingredients, pictureUrl
+    frenchRecipes.push(
+      new Recipe(
+        recipie.id,
+        recipie.name,
+        recipie.description,
+        recipie.ingredients,
+        recipie.pictureUrl
+      )
+    );
+  });
```

Comment:
As an alternative way to populate the `frechRecipes` array, you can use [Array's `map` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map). It's a bit more advanced though, but I figured it might be interesting to learn about it. The whole idea of the `map` is that it takes every element of an array, transforms it and then returns the new array with transformed elements. 

You don't have to change your code as it works well, just wanted to provide a small tip 😄 



```suggestion
  frenchRecipes = fetchedRecipes.map((recipie) => {
    return (
      new Recipe(
        recipie.id,
        recipie.name,
        recipie.description,
        recipie.ingredients,
        recipie.pictureUrl
      )
    );
  });
```